functionary/builder/utils.py

```diff
@@ -105,6 +107,40 @@ def initiate_build(
     return build
 
 
+def _format_log_string(build_results, push_results):
+    """
+    Helper function to transform results of docker
+    push and docker build into more user friendly string.
+
+    Args:
+        build_results: itertools._tee object returned from docker build
+        push_results: generator object returned from docker push
+
+    Returns:
+        string with both build and push information in more readable format
+    """
+    build_str = ""
+    for line in build_results:
+        line_str = ""
+        for value in line.values():
+            line_str = line_str + str(value) + " "
```

Comment:
It looks like your intention is to have a space before each line, so just a heads up that as is you don't end up with a space before the first line.  You can fix that by injecting the space in the next line, similar to what you do with the `push_str`:

```python
line_str = line_str + str(value)
build_str = build_str + " " + line_str
```